network/src/web.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  UnsupportedBrowserException,
+  WebPlugin,
+  PluginListenerHandle,
+} from '@capacitor/core';
+import {
+  NetworkPlugin,
+  NetworkStatus,
+  NetworkStatusConnectionType,
+} from './definitions';
+
+declare var window: any;
+
+function translateConnection(connection: any): NetworkStatusConnectionType {
+  let result: NetworkStatusConnectionType = 'unknown';
+  const type = connection ? connection.type || connection.effectiveType : null;
+  if (type && typeof type === 'string') {
+    switch (type) {
+      // possible type values
+      case 'bluetooth':
+      case 'cellular':
+        result = 'cellular';
+        break;
+      case 'none':
+        result = 'none';
+        break;
+      case 'ethernet':
+      case 'wifi':
+      case 'wimax':
+        result = 'wifi';
+        break;
+      case 'other':
+      case 'unknown':
+        result = 'unknown';
+        break;
+      // possible effectiveType values
+      case 'slow-2g':
+      case '2g':
+      case '3g':
+        result = 'cellular';
+        break;
+      case '4g':
+        result = 'wifi';
+        break;
+      default:
+        break;
+    }
+  }
```

Comment:
[`type`](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/type) and [`effectiveType`](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/effectiveType) reference

This is a best effort to map the possible values onto the ones we are documented to returned. Since the native side doesn't have all of these possibilities, it didn't make sense to return them as-is.